src/pcre2test.c

```diff
@@ -2943,12 +2944,60 @@ return sys_errlist[n];
 *            Local memory functions              *
 *************************************************/
 
+static int mallocs_until_failure = INT_MAX;
+static int mallocs_called = 0;
+
 /* Alternative memory functions, to test functionality. */
 
 static void *my_malloc(size_t size, void *data)
 {
-void *block = malloc(size);
+void *block;
+
 (void)data;
+
+
+/*
+XXX
+
+list of all malloc'ing functions:
+
+FIXED COUNT:
+,pcre2_code_copy
+,pcre2_code_copy_with_tables
+,pcre2_compile_context_copy_8,__libc_start_main,_start,
+,pcre2_convert_context_copy_8,__libc_start_main,_start,
+,pcre2_general_context_copy_8,__libc_start_main,_start,
+,pcre2_match_context_copy_8,__libc_start_main,_start,
+,pcre2_compile_context_create_8,__libc_start_main,_start,
+,pcre2_convert_context_create_8,__libc_start_main,_start,
+,pcre2_general_context_create_8,__libc_start_main,_start,
+,pcre2_match_context_create_8,__libc_start_main,_start,
+,pcre2_general_context_create_8,__libc_start_main,_start,
+
+,pcre2_maketables_8,__libc_start_main,_start,
+,pcre2_match_data_create_8,__libc_start_main,_start,
+,pcre2_match_data_create_from_pattern_8,__libc_start_main,_start,
+,pcre2_pattern_convert_8,__libc_start_main,_start,
+
+,pcre2_serialize_decode_8,__libc_start_main,_start,
+,pcre2_serialize_encode_8,__libc_start_main,_start,
+,pcre2_substring_get_bynumber_8,__libc_start_main,_start,
+,pcre2_substring_get_byname_8,__libc_start_main,_start,
+,pcre2_substring_list_get_8,__libc_start_main,_start,
+
+// XXX VARIABLE:
+,pcre2_match_8,__libc_start_main,_start,
+,pcre2_dfa_match_8,__libc_start_main,_start,
+,pcre2_substitute_8,__libc_start_main,_start,
```

Comment:
TODO
- [ ] pcre2_substitute
- All the ones above which don't need a loop - which do a fixed number of malloc calls
  - [ ] code_copy[_with_tables]
  - [ ] context_create/context_copy functions
  - [ ] maketables
  - [ ] pattern_convert
  - [ ] serialize_{encode,decode}
  - [ ] substring_{get,list}